src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+
+// import VirtualMachineInstanceModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineInstanceModel';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+// import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Modal,
+  ModalVariant,
+  NumberInput,
+  Title,
+  TitleSizes,
+} from '@patternfly/react-core';
+
+import useTemplateDefaultCpuMemory from './hooks/useTemplateDefaultCpuMemory';
+import { getCPUCount, getMemorySize, memorySizesTypes } from './utils/CpuMemoryUtils';
+
+import './cpu-memory-modal.scss';
+
+type FlavorModalProps = {
+  vmi: V1VirtualMachineInstance;
+};
+
+const CPUMemoryModal: React.FC<FlavorModalProps> = ({ vmi }) => {
+  const { t } = useKubevirtTranslation();
+  const [showFlavorModal, setShowFlavorModal] = React.useState(true);
+  const [data, loaded, error] = useTemplateDefaultCpuMemory(vmi);
+
+  const [memory, setMemory] = React.useState<number>();
+  const [cpu, setCpu] = React.useState<number>();
+  const [memoryUnit, setMemoryUnit] = React.useState<string>();
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    if (vmi?.metadata) {
+      const requests = vmi?.spec?.domain?.resources?.requests as { [key: string]: string };
+      const { size, unit } = getMemorySize(requests?.memory);
+      setMemoryUnit(unit);
+      setMemory(size);
+      setCpu(getCPUCount(vmi?.spec?.domain?.cpu));
+    }
+  }, [vmi]);
+
+  return (
+    <Modal
+      title={t('Edit CPU | Memory')}
+      isOpen={showFlavorModal}
+      className="cpu-memory-modal"
+      variant={ModalVariant.small}
+      onClose={() => setShowFlavorModal(false)}
+      actions={[
+        <Button
+          key="confirm"
+          variant={ButtonVariant.primary}
+          onClick={() => {
+            return null;
+            // k8sUpdate<V1VirtualMachineInstance>({
```

Comment:
this is why we introduced `immer` so we dont need to care about untouched parts.

delete this for now